### PR TITLE
Accept multiple requests from yaml templates

### PR DESCRIPTION
**Pull request by jronak**
Yaml template can be used to send gRPC stream requests now, `requests` field accepts multiple requests.

Example template: 
```
service: foo
procedure: Bar/BidiStream
caller: bar
timeout: 4.5s
requests:
    - test: 1
    - test: 2
```

Template command:
```
yab -y test/stream.yab -p grpc://localhost:55555
{
  "value": "1"
}

{
  "value": "12"
}
```

### CODE REVIEW

File: template_test.go [hunk 1]
```diff
@@ -81,6 +81,13 @@ func TestTemplate(t *testing.T) {
 	assert.True(t, opts.ROpts.ThriftDisableEnvelopes)
 }
 
+func TestTemplateRequestsField(t *testing.T) {
+	opts := newOptions()
+	mustReadYAMLFile(t, "testdata/templates/stream.yab", opts)
+
+	assert.Equal(t, "---\ntest: 1\n---\ntest: 2\n", opts.ROpts.RequestJSON)
```
Comment (Dogild) [inline template_test.go:88]: nit: Let's have the following format (easier to read) for the test

```
want = `---
test: 1
---
test: 2
```

assert.Equal(t, want, opts.ROpts.RequestJSON)

(off commit) having `opts.ROpts.RequestJSON` and not json as value is odd

File: template.go [hunk 2]
```diff
@@ -79,19 +80,42 @@ func readYAMLFile(yamlTemplate string, templateArgs map[string]string, opts *Opt
...
+	if t.Request != nil {
+		req, err := templateargs.ProcessMap(t.Request, templateArgs)
+		if err != nil {
+			return nil, err
+		}
+
+		return yaml.Marshal(req)
+	}
+
+	var body string
```
Comment (Dogild) [inline template.go:93]: nit: you could use a string builder here (faster for string concatenation). This code is not in a hot path, so only nit

**Review by Dogild — APPROVED** [review]

File: template.go [hunk 3]
```diff
@@ -47,11 +47,12 @@ type template struct {
...
-	Headers map[string]string           `yaml:"headers"`
-	Baggage map[string]string           `yaml:"baggage"`
-	Jaeger  bool                        `yaml:"jaeger"`
-	Request map[interface{}]interface{} `yaml:"request"`
-	Timeout time.Duration               `yaml:"timeout"`
+	Headers  map[string]string             `yaml:"headers"`
+	Baggage  map[string]string             `yaml:"baggage"`
+	Jaeger   bool                          `yaml:"jaeger"`
+	Request  map[interface{}]interface{}   `yaml:"request"`
+	Requests []map[interface{}]interface{} `yaml:"requests"`
```
Comment (prashantv) [inline template.go:54]: what happens if both are set? should we return an error (I think we ignore `Request` silently right now)

File: template.go (hunk 2 above)
Comment (jronak) [reply template.go]: good one, this didn't require string use actually so changed body to `[]byte` which now uses `append`

File: template_test.go (hunk 1 above)
Comment (jronak) [reply template_test.go]: Yup `RequestJSON` weirdly carries YAML as well. This needs some cleanup for sure

File: template.go (hunk 3 above)
Comment (jronak) [reply template.go:57]: I missed adding this check earlier, now it returns error if both fields are set

**Review by prashantv — APPROVED** [review]

**Pull request merged**